src/script/common/c_content.cpp

```diff
@@ -2033,11 +2033,11 @@ bool read_tree_def(lua_State *L, int idx, const NodeDefManager *ndef,
 	getboolfield(L, idx, "thin_branches", tree_def.thin_branches);
 	tree_def.fruit_chance = 0;
 	getstringfield(L, idx, "fruit", fruit);
-	if (!fruit.empty()) {
+	if (fruit.empty())
+		fruit = "air";
```

Comment:
Why do we need this branch?
```C++
	tree_def.fruit_chance = 0;
	getstringfield(L, idx, "fruit", fruit);
	getintfield(L, idx, "fruit_chance", tree_def.fruit_chance);
```
seems to work as well.

Reply:
As I wrote, to not trigger a node resolver error when `fruit` is left empty.
e.g. try this:
``` lua
core.register_chatcommand("t", {
    func = function(name)
        local player = core.get_player_by_name(name)
        core.spawn_tree(player:get_pos(), {
		axiom = "TT",
		trunk = "mapgen_tree",
		leaves = "mapgen_stone"
	})
    end,
})
```

Alternatively, you would have to tell the node resolver somehow to not resolve the node.
(previously `fruit_chance` was used, but this caused the bug)

Reply:
I see. Whereas there is a fallback option in `getIdFromNrBacklog`, we cannot differentiate between "no value" and "invalid node" there. That's unfortunate.
In this case I'd recommend to set `fruit = "air"` before `getstringfield`. Lua-provided empty strings would then get caught by the node resolver and logged appropriately.

Reply:
Oh right, I changed it, now `fruit = ""` does not default to `fruit = "air"` and you get any error by the node resolver instead.